### PR TITLE
[frontend] #2030 ember-way for statistics loading logic

### PR DESCRIPTION
**Pull request by yrakova**
[#2030 ] [frontend] Backport admin statistics to client app
    
* using serializers/adapters for parsing statistics data
* fix tests accordingly

### CODE REVIEW

File: frontend/app/components/statistics/component.ts [hunk 1]
```diff
@@ -7,47 +7,74 @@ import { DateTime } from 'luxon';
 import { tracked } from '@glimmer/tracking';
 import { task, Task as TaskGenerator } from 'ember-concurrency';
 import { action } from '@ember/object';
+import Store from '@ember-data/store';
 
-export default class StatisticsComponent extends Component {
+function formatDateForRequest(date: DateTime): string {
```
Comment (lifeart) [inline frontend/app/components/statistics/component.ts:12]: this logic should be in serializer

File: frontend/app/components/statistics/component.ts [hunk 2]
```diff
@@ -7,47 +7,74 @@ import { DateTime } from 'luxon';
...
+    const fromYear: DateTime = this.selectedMonth.startOf('year');
+    const toYear: DateTime = this.selectedMonth.endOf('year');
     this.isLoadingMonthTimeTrackData = true;
 
-    this.monthTimeTrackData = yield this.network.getUserStatisticsByYear(
-      fromYear,
-      toYear,
-    );
+    const from: string = formatDateForRequest(fromYear);
+    const to: string = formatDateForRequest(toYear);
```
Comment (lifeart) [inline frontend/app/components/statistics/component.ts:59]: should be in serializer

File: frontend/app/components/statistics/component.ts [hunk 3]
```diff
@@ -7,47 +7,74 @@ import { DateTime } from 'luxon';
...
+
+    this.monthTimeTrackData = yield this.store
+      .query('user-yearly-statistics', {
+        from,
+        to,
+      })
+      .catch((error) => {
+        console.error(error);
```
Comment (lifeart) [inline frontend/app/components/statistics/component.ts:67]: no promse-catch in yield, use { try { .. } catch (e) { ... } notation

File: frontend/app/routes/profile/statistics.ts [hunk 4]
```diff
@@ -1,4 +1,8 @@
 import Route from '@ember/routing/route';
+import { DateTime } from 'luxon';
 
 export default class StatisticsRoute extends Route {
-}
\ No newline at end of file
+  get initialSelectedMonth(): DateTime {
+    return DateTime.now();
```
Comment (lifeart) [inline frontend/app/routes/profile/statistics.ts:6]: wondering, should it be on controller level?

**Review by lifeart — APPROVED** [review]

File: frontend/app/components/statistics/component.ts (hunk 1 above)
Comment (yrakova) [reply frontend/app/components/statistics/component.ts]: Done

File: frontend/app/components/statistics/component.ts (hunk 2 above)
Comment (yrakova) [reply frontend/app/components/statistics/component.ts]: Done

File: frontend/app/routes/profile/statistics.ts (hunk 4 above)
Comment (yrakova) [reply frontend/app/routes/profile/statistics.ts]: Sure

File: frontend/app/components/statistics/component.ts (hunk 3 above)
Comment (yrakova) [reply frontend/app/components/statistics/component.ts]: Done

File: frontend/app/transforms/full-date.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+// eslint-disable-next-line ember/use-ember-data-rfc-395-imports
+import DS from 'ember-data';
+import { DateTime } from 'luxon';
+
+const { DateTransform } = DS;
+const FullDate = DateTransform.extend({
+  deserialize(serialized: Record<string, never>): DateTime | null {
+    return (serialized && DateTime.fromISO(serialized)) || null;
```
[the remark this answers is not in the document]
Comment (yrakova) [reply frontend/app/transforms/full-date.ts]: Done

**Review by lifeart — APPROVED** [review]

**Comment by lifeart** [comment]
@yrakova thank you for contribution!

**Pull request merged**